wpilibj/src/main/java/edu/wpi/first/wpilibj/geometry/Pose2dSource.java

```diff
@@ -0,0 +1,122 @@
+  private final AngleMeasurement m_angleMeasurement;
+  private final DistanceMeasurement m_distanceMeasurement;
+  private final Supplier<Pose2d> m_source;
+
+  /**
+   * Creates a {@code Pose2dSource} that wraps the given source.
+   * @param source the source of the poses, i.e the odometry's {@link
+   *  edu.wpi.first.wpilibj.kinematics.DifferentialDriveOdometry#getPoseMeters() getPose()} method.
+   * @param name the name to associate with this Pose2dSource in LiveWindow.
+   Defaults to {@code "Robot Pose"}.
+   */
+  public Pose2dSource(Supplier<Pose2d> source, String name) {
+    this(source, name, AngleMeasurement.kDegrees);
+  }
+
+  /**
+   * Creates a {@code Pose2dSource} that wraps the given source.
+   * @param source the source of the poses, i.e the odometry's {@link
+   *  edu.wpi.first.wpilibj.kinematics.DifferentialDriveOdometry#getPoseMeters() getPose()} method.
+   * @param name the name to associate with this Pose2dSource in LiveWindow.
+   Defaults to {@code "Robot Pose"}.
+   * @param distanceMeasurement the units that the pose is shown in LiveWindow (inches/meters).
+   * @param angleMeasurement the units that the rotation is shown in LiveWindow (degrees/radians).
+   */
+  public Pose2dSource(Supplier<Pose2d> source, String name, DistanceMeasurement distanceMeasurement,
+                      AngleMeasurement angleMeasurement) {
+    m_source = source;
+    m_distanceMeasurement = distanceMeasurement;
+    m_angleMeasurement = angleMeasurement;
+
+    SendableRegistry.addLW(this, name);
+  }
+
+  /**
+   * Creates a {@code Pose2dSource} that wraps the given source.
+   * @param source the source of the poses, i.e the odometry's {@link
+   *  edu.wpi.first.wpilibj.kinematics.DifferentialDriveOdometry#getPoseMeters() getPose()} method.
+   */
+  public Pose2dSource(Supplier<Pose2d> source) {
+    this(source, "Robot Pose");
+  }
+
+  /**
+   * Creates a {@code Pose2dSource} that wraps the given source.
+   * @param source the source of the poses, i.e the odometry's {@link
+   *  edu.wpi.first.wpilibj.kinematics.DifferentialDriveOdometry#getPoseMeters() getPose()} method.
+   * @param name the name to associate with this Pose2dSource in LiveWindow.
+  Defaults to {@code "Robot Pose"}.
+   * @param angleMeasurement the units that the rotation is shown in LiveWindow (degrees/radians).
+   */
+  public Pose2dSource(Supplier<Pose2d> source, String name, AngleMeasurement angleMeasurement) {
+    this(source, name, DistanceMeasurement.kMeters, angleMeasurement);
+  }
+
+  @Override
+  public void initSendable(SendableBuilder builder) {
+    builder.setActuator(false);
+    builder.addDoubleProperty(String.format("x_%s", m_distanceMeasurement),
+        () -> get().getTranslation().getX(), null);
+    builder.addDoubleProperty(String.format("y_%s", m_distanceMeasurement),
+        () -> get().getTranslation().getY(), null);
+    builder.addDoubleProperty(String.format("rotation_%s", m_angleMeasurement),
```

Comment:
Would this add NetworkTable entry keys x_meters, y_meters, and rotation_degrees? I think adding the unit is unnecessary and goes against the current convention of LW which is to send the value and have the consumer (e.g. shuffleboard) make the conversions. Take the [AnalogAccelerometer](https://github.com/wpilibsuite/allwpilib/blob/master/wpilibj/src/main/java/edu/wpi/first/wpilibj/AnalogAccelerometer.java#L142), [GyroBase](https://github.com/wpilibsuite/allwpilib/blob/master/wpilibj/src/main/java/edu/wpi/first/wpilibj/GyroBase.java#L56), or [Ultrasonic](https://github.com/wpilibsuite/allwpilib/blob/master/wpilibj/src/main/java/edu/wpi/first/wpilibj/Ultrasonic.java#L412) classes. They each send keys without the name of the unit appended to them. 

It also seems like the snake_case format goes against their naming conventions of capitalizing and having spaces between each word. I think the correct way here is to send the keys **X**, **Y**, and **Rotation**, and send the values as meters and degrees (degrees since that's what gyros are sending their values as to LW).

Reply:
I agree with this. I don't think we need units at all in this because the Pose2d itself doesn't store any data about what kind of units are being used. Heck, it's valid to use a Pose2d to represent a position in miles.

Reply:
Removed units. But seriously, can someone please run the testbench?